### ISSUE TITLE
ALG_SM3_256 cannot be enabled with OpenSSL 3.x

### ISSUE THREAD

**Issue by gotthardp** [kgoldman/ibmswtpm2#18]
Enabling `ALG_SM3_256` causes a build error on OpenSSL 3.x:
```
TpmToOsslHash.h:77:14: fatal error: openssl/sm3.h: No such file or directory
   77 | #    include <openssl/sm3.h>
      |              ^~~~~~~~~~~~~~~
```
This is because the `sm3.h` is an internal file, which is not available on standard installations.

In fact, all `xxx_Init`, `xxx_Update` and `xxx_Final` functions listed in `src/TpmToOsslHash.h` are deprecated since OpenSSL 3.0.0, see https://docs.openssl.org/3.2/man3/SHA256_Init.

I suggest to rewrite the hashes using `EVP_DigestInit`, `EVP_DigestUpdate` and `EVP_DigestFinal`. These are available for a quite long time and compatible with both OpenSSL 1.1.1x and OpenSSL 3.x.

**Comment by kgoldman** [comment]
I agree.  Most of the crypto code came from the TPM specification.  I asked on the TPM WG.

Do you have a need for sm3? If so, and TCG does not want to implement it, I can do so. Do you have test cases? Otherwise I'd have to create some before I merge a patch.

For EVP_, I've changed the TSS side. Since the old commands are just deprecated, not deleted, I'd rather leave them because the TCG implementation uses deeply nested macros to support other crypto providers. It's not an easy replacement.
